migrations/20240624145325_recreate_auth_functions.up.sql

```diff
@@ -0,0 +1,50 @@
+-- set the search_path to an empty string to force fully qualified names in the function 
+do $$ 
+begin
+    -- auth.uid() function
+    create or replace function {{ index .Options "Namespace" }}.uid() 
+        returns uuid 
+        set search_path to '' 
+    as $func$
+      select
+        coalesce(
+          nullif(current_setting('request.jwt.claim.sub', true), ''),
+          (nullif(current_setting('request.jwt.claims', true), '')::jsonb ->> 'sub')
+        )::uuid
+    $func$ language sql stable;
+
+    -- auth.role() function
+    create or replace function {{ index .Options "Namespace" }}.role() 
+        returns text 
+        set search_path to ''
+    as $func$
+      select
+        coalesce(
+          nullif(current_setting('request.jwt.claim.role', true), ''),
+          (nullif(current_setting('request.jwt.claims', true), '')::jsonb ->> 'role')
+        )::text
+    $func$ language sql stable;
+
+    -- auth.email() function
+    create or replace function {{ index .Options "Namespace" }}.email()
+        returns text
+        set search_path to ''
+    as $func$
+      select
+        coalesce(
+          nullif(current_setting('request.jwt.claim.email', true), ''),
+          (nullif(current_setting('request.jwt.claims', true), '')::jsonb ->> 'email')
+        )::text
+    $func$ language sql stable;
+
+    -- auth.jwt() function
+    create or replace function {{ index .Options "Namespace" }}.jwt()
+        returns jsonb
+    as $func$
+      select
+        coalesce(
+          nullif(current_setting('request.jwt.claim', true), ''),
+          nullif(current_setting('request.jwt.claims', true), '')
```

Comment:
Hmm is it like this?

Reply:
Are you referring to the lack of `search_path` ? Set that to '' the rest should be accurate

Reply:
Not actually sure that we need to replace `.jwt()` at all let me check.

Reply:
Sorry just wanted to validate this... https://github.com/supabase/auth/blob/e81c25d19551fdebfc5197d96bc220ddb0f8227b/migrations/20220531120530_add_auth_jwt_function.up.sql#L7